### PR TITLE
[iTunesLibrary] Xcode 10 beta 3 support.

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

File: src/iTunesLibrary/Enums.cs [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+		TVShows = 2,
+		Music = 3,
+		Audiobooks = 4,
+		Books = Audiobooks,
+		Ringtones = 5,
+		Podcasts = 7,
+		VoiceMemos = 14,
+		Purchases = 16,
+		iTunesU = 26,
+		ITLibDistinguishedPlaylistKind90sMusic = 42,
```
Comment (chamons) [inline src/iTunesLibrary/Enums.cs]: Should this just be Type90sMusic?

File: src/iTunesLibrary/Enums.cs [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+		TVShow = 8,
+		InteractiveBooklet = 9,
+		HomeVideo = 12,
+		Ringtone = 14,
+		DigitalBooklet = 15,
+		IOSApplication = 16,
+		VoiceMemo = 17,
+		iTunesU = 18,
+		Book = 19,
+		PDFBook = 20,
```
Comment (rolfbjarne) [inline src/iTunesLibrary/Enums.cs]: `Pdf`

File: src/ituneslibrary.cs [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+		nuint TrackCount { get; }
+
+		[NullAllowed, Export ("albumArtist")]
+		string AlbumArtist { get; }
+
+		[NullAllowed, Export ("sortAlbumArtist")]
+		string SortAlbumArtist { get; }
+
+		[Export ("persistentID", ArgumentSemantic.Retain)]
+		NSNumber PersistentID { get; }
```
Comment (rolfbjarne) [inline src/ituneslibrary.cs]: `Id`

File: src/ituneslibrary.cs [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+	interface ITLibArtist
+	{
+		[NullAllowed, Export ("name")]
+		string Name { get; }
+
+		[NullAllowed, Export ("sortName")]
+		string SortName { get; }
+
+		[Export ("persistentID", ArgumentSemantic.Retain)]
+		NSNumber PersistentID { get; }
```
Comment (rolfbjarne) [inline src/ituneslibrary.cs]: `Id`

File: src/ituneslibrary.cs [hunk 5]
```diff
@@ -0,0 +1,388 @@
...
+	}
+
+	delegate void ITLibMediaEntityEnumerateValuesHandler (NSString property, NSObject value, out bool stop);
+
+	[Mac (10,14, onlyOn64: true)]
+	[BaseType (typeof(NSObject))]
+	interface ITLibMediaEntity
+	{
+		[Export ("persistentID", ArgumentSemantic.Retain)]
+		NSNumber PersistentID { get; }
```
Comment (rolfbjarne) [inline src/ituneslibrary.cs]: `Id`

File: src/ituneslibrary.cs [hunk 6]
```diff
@@ -0,0 +1,388 @@
...
+
+		[Export ("valueForProperty:")]
+		[return: NullAllowed]
+		NSObject GetValue (string property);
+
+		[Export ("enumerateValuesForProperties:usingBlock:")]
+		void EnumerateValues (NSSet<NSString> properties, ITLibMediaEntityEnumerateValuesHandler handler);
+
+		[Export ("enumerateValuesExceptForProperties:usingBlock:")]
+		void EnumerateValuesExcluding (NSSet<NSString> properties, ITLibMediaEntityEnumerateValuesHandler handler);
```
Comment (rolfbjarne) [inline src/ituneslibrary.cs]: IMHO `EnumerateValuesExcept` sounds just as good as `EnumerateValuesExcluding`, and it has the additional benefit of being closer to the ObjC name.

File: src/ituneslibrary.cs [hunk 7]
```diff
@@ -0,0 +1,388 @@
...
+	interface ITLibPlaylist
+	{
+		[Export ("name")]
+		string Name { get; }
+
+		[Export ("master")]
+		bool Master { [Bind ("isMaster")] get; }
+
+		[NullAllowed, Export ("parentID", ArgumentSemantic.Retain)]
+		NSNumber ParentID { get; }
```
Comment (rolfbjarne) [inline src/ituneslibrary.cs]: `Id`

File: src/iTunesLibrary/Enums.cs [hunk 8]
```diff
@@ -0,0 +1,321 @@
...
+		Podcast = 4,
+		Audiobook = 5,
+		PdfBooklet = 6,
+		MusicVideo = 7,
+		TVShow = 8,
+		InteractiveBooklet = 9,
+		HomeVideo = 12,
+		Ringtone = 14,
+		DigitalBooklet = 15,
+		IOSApplication = 16,
```
Comment (spouliot) [inline src/iTunesLibrary/Enums.cs]: -> `iOSApplication`
like `iTunesU` :)

File: src/iTunesLibrary/Enums.cs [hunk 9]
```diff
@@ -0,0 +1,321 @@
...
+		[Field ("ITLibPlaylistPropertyItems")]
+		Items,
+		[Field ("ITLibPlaylistPropertyKind")]
+		Kind,
+	}
+
+	[Mac (10,14, onlyOn64: true)]
+	public enum ITLibMediaEntityProperty {
+		[Field ("ITLibMediaEntityPropertyPersistentID")]
+		PersistentID,
```
Comment (spouliot) [inline src/iTunesLibrary/Enums.cs]: -> `Id`

File: src/ituneslibrary.cs [hunk 10]
```diff
@@ -0,0 +1,390 @@
...
+		nuint Season { get; }
+
+		[NullAllowed, Export ("episode")]
+		string Episode { get; }
+
+		[Export ("episodeOrder")]
+		nint EpisodeOrder { get; }
+
+		[Export ("hd")]
+		bool Hd { [Bind ("isHD")] get; }
```
Comment (spouliot) [inline src/ituneslibrary.cs]: two letters acronym are fine, i.e. `HD`

**Pull request merged**